File: results/ChatGPT/python_streamlit/greet.py
import streamlit as st

# Define a dictionary of greetings in different languages
greetings = {
    'English': 'Hello',
    'French': 'Bonjour',
    'Spanish': 'Hola'
}

# Create a dropdown selection box for language
language = st.selectbox('Select a language', ('English', 'French', 'Spanish'))

# Create a text input field for name
name = st.text_input('Enter your name')

# Create a button to greet the user
if st.button('Greet'):
    # Get the greeting based on the selected language
    greeting = greetings[language]
    # Display the greeting with the user's name
    st.write(f'{greeting}, {name}!')

File: results/ChatGPT/javascript_React/greet.js
import React, { useState } from 'react';

const Greeting = () => {
  const [name, setName] = useState('');
  const [language, setLanguage] = useState('English');
  const [greeting, setGreeting] = useState('');

  const handleNameChange = (event) => {
    setName(event.target.value);
  };

  const handleLanguageChange = (event) => {
    setLanguage(event.target.value);
  };

  const handleGreetClick = () => {
    switch (language) {
      case 'English':
        setGreeting(`Hello ${name}!`);
        break;
      case 'French':
        setGreeting(`Bonjour ${name}!`);
        break;
      case 'Spanish':
        setGreeting(`Hola ${name}!`);
        break;
      default:
        setGreeting('');
    }
  };

  return (
    <div>
      <label htmlFor="name">Name:</label>
      <input type="text" id="name" value={name} onChange={handleNameChange} />
      <br />
      <label htmlFor="language">Language:</label>
      <select id="language" value={language} onChange={handleLanguageChange}>
        <option value="English">English</option>
        <option value="French">French</option>
        <option value="Spanish">Spanish</option>
      </select>
      <br />
      <button onClick={handleGreetClick}>Greet</button>
      <br />
      <p>{greeting}</p>
    </div>
  );
};

export default Greeting;